### PR TITLE
React/DP-12414: Responsive Table.

### PR DESCRIPTION
**Pull request by smurrayatwork**
## Description
This updates the Table component for it to support responsiveness. 

## Related Issue / Ticket

- [JIRA issue](https://jira.mass.gov/browse/DP-12414)

## Steps to Test
1. Pull the branch and browse to atoms -> table-> table. Verify the Table works as intended for the ticket.

## Screenshots
screencapture-localhost-6006-iframe-html-2019-02-01-12_23_13

### CODE REVIEW

**Review by avrilpearl — APPROVED** [review]

File: react/src/components/atoms/table/Table/index.js [hunk 1]
```diff
@@ -11,68 +12,84 @@ const Table = (table) => {
...
+            }
+          </tbody>))
+          }
+        </table>
+      </div>
+    </div>
   );
 };
 
 Table.propTypes = {
```
Comment (avrilpearl) [inline react/src/components/atoms/table/Table/index.js]:
```suggestion
Table.PropTypes = {
```

File: react/src/components/atoms/table/Table/index.js [hunk 2]
```diff
@@ -11,68 +12,84 @@ const Table = (table) => {
...
-              <tr key={row.key || defaultRowKey} className={row.rowSpanOffset ? 'is-offset' : null}>
-                { row.cells.map((cell, cellIndex) => {
-                  let dataLabel = null;
-                  if (tableHeaderRow.length > 0) {
-                    // Mayflower twig loops started at 1 - add one to index.
-                    dataLabel = tableHeaderRow.cells[(cellIndex + 1) - tableIndexOffset].text;
-                  }
-                  const defaultCellKey = `${defaultRowKey}-cell${cellIndex}`;
-                  return(cell.heading ? (
+    <div className="ma__table--responsive js-ma-responsive-table">
```
Comment (clairesunstudio) [inline react/src/components/atoms/table/Table/index.js]: this wrapper doesn't seem neccessary

File: react/src/components/atoms/table/Table/index.js [hunk 3]
```diff
@@ -11,68 +12,84 @@ const Table = (table) => {
...
+      <div className={responsiveClasses} role="region" id={table.id}>
```
Comment (clairesunstudio) [inline react/src/components/atoms/table/Table/index.js]: same as above

File: react/src/components/atoms/table/Table/style.scss [hunk 4]
```diff
@@ -1,2 +1,9 @@
 @import 'global';
-@import '01-atoms/table';
\ No newline at end of file
+@import '01-atoms/table';
+@import '01-atoms/table-responsive';
```
Comment (clairesunstudio) [inline react/src/components/atoms/table/Table/style.scss]: this scss import is not necessary

File: react/src/components/atoms/table/Table/index.js [hunk 5]
```diff
@@ -11,68 +12,84 @@ const Table = (table) => {
...
+        }{table.bodies.map((body, bodyIndex) => (
+          <tbody key={body.key || `tbody${bodyIndex}`}>
+            { body.rows.map((row, rowIndex) => {
+              const tableIndexOffset = row.rowSpanOffset ? 0 : 1;
+              const defaultRowKey = `tbody${bodyIndex}-row${rowIndex}`;
+              return(
+                <tr key={row.key || defaultRowKey} className={row.rowSpanOffset ? 'is-offset' : null}>
+                  { row.cells.map((cell, cellIndex) => {
+                    let dataLabel = null;
+                    if ((cellIndex > 0) && tableHeaderRow.cells && tableHeaderRow.cells.length > 0) {
```
Comment (clairesunstudio) [inline react/src/components/atoms/table/Table/index.js]: tableHeaderRow is not defined?

File: react/src/components/atoms/table/Table/index.js [hunk 6]
```diff
@@ -11,68 +12,84 @@ const Table = (table) => {
...
+      </div>
+    </div>
   );
 };
 
 Table.propTypes = {
+  id: PropTypes.string,
   head: PropTypes.shape({
     rows: PropTypes.arrayOf(PropTypes.shape({
       rowSpanOffset: PropTypes.bool,
```
Comment (clairesunstudio) [inline react/src/components/atoms/table/Table/index.js:90]: can you add a description here for the rowSpanOffset. Currently throwing an error if true in bodies.rows, and I don't think it's doing anything in head.rows

File: react/src/components/atoms/table/Table/index.js (hunk 5 above)
Comment (smurrayatwork) [reply react/src/components/atoms/table/Table/index.js]: It's defined on line 10

File: react/src/components/atoms/table/Table/index.js (hunk 6 above)
Comment (smurrayatwork) [reply react/src/components/atoms/table/Table/index.js:90]: The initial example already uses it with it some things being true/false, and there's no error.

File: react/src/components/atoms/table/Table/index.js (hunk 5 above)
Comment (clairesunstudio) [reply react/src/components/atoms/table/Table/index.js]: oh missed that, i'm seeing this error when i change rowSpanOffset to true on line 30 in Table.knobs.options.js
screen shot 2019-02-06 at 11 23 42 am

**Comment by clairesunstudio** [comment]
looks good! need to follow up with rowSpanOffset check to prevent hard error and remove that from rows

**Comment by clairesunstudio** [comment]
screen shot 2019-02-06 at 3 07 10 pm
fixed the bug not showing the first row label ^ as well as preventing hard error

**Pull request merged**